src/pkg/caendr/caendr/models/datastore/cart.py

```diff
@@ -0,0 +1,93 @@
+from caendr.models.datastore import Entity
+from caendr.models.error import NotFoundError, NonUniqueEntity
+from caendr.utils.data import unique_id
+
+class PRICES:
+  DIVERGENT_SET = 160
+  STRAIN_SET = 640
+  STRAIN = 15
+  SHIPPING = 65
+
+
+
+class Cart(Entity):
+  kind = 'cart'
+
+  def __init__(self, name_or_obj = None, *args, **kwargs): # If nothing passed for name_or_obj, create a new ID to use for this object
+    if name_or_obj is None: 
+      name_or_obj = unique_id()
+      self.set_properties_meta(id = name_or_obj) 
+
+  # Initialize from superclass 
+    super().__init__(name_or_obj, *args, **kwargs)
+    if not self['is_deleted']:
+      self['is_deleted'] = False
+    if not self['items']:
+      self['items'] = []
+  
+  ## Properties List ##
+
+  @classmethod
+  def get_props_set(cls):
+    return {
+      *super().get_props_set(),
+      'user',
+      'items',
+      'is_deleted'
+    }
+  
+  
+
+  @classmethod
+  def lookup_by_user(cls, email):
+
+    # Try finding and returning unique card for this user
+    try:
+      return Cart.query_ds_not_deleted('user', email, required=True)
+
+
+    # User doesn't have a cart
+    except NotFoundError:
+      return Cart(**{'user': email, 'items': []})
+
+    # Multiple carts found for user
+    # TODO: get all carts that match the user, sort by date created on and return the latest one?
+    except NonUniqueEntity:
+      return 'multiple carts found'
+
+
+  def transfer_to_user(self, email):
+    self['user'] = email
+
+
+  def add_item(self, item):
+    for cartItem in self['items']:
+      if cartItem['name'] == item['name']:
+        return
+    self['items'].append(item)
+
+
+  def remove_item(self, item):
+    for cartItem in self['items']:
+      if cartItem['name'] == item:
+        self['items'].remove(cartItem)
+        return
+      else:
+        continue  
+
+
+  def soft_delete(self):
+    self['is_deleted'] = True
+
+
+  def get_price(self, item):
```

Comment:
Same note as above -- this should probably be a class method.

Reply:
Completely optional extension to this -- you could actually move this to the `PRICES` class, if you wanted to.  I'd recommend something like:
```
class PRICE:
  
  # insert price variables

  def lookup(item):
    # return the price
```

Then you could use it as:
```
  item['price'] = PRICE.lookup(item)
```

This is totally up to you.  I personally think it's cleaner, since it bundles the pricing logic together in one place, but it's not like it fixes a potential bug or anything.